bundles/rust/rust_lexer.moon

```diff
@@ -0,0 +1,93 @@
+--- rust lexer for howl from [ta-rust](https://bitbucket.org/a_baez/ta-rust)
+-- See @{README.md} for details on usage.
+-- @author [Alejandro Baez](https://keybase.io/baez)
+-- @copyright 2016
+-- @license MIT (see LICENSE)
+-- @module rust
+
+howl.util.lpeg_lexer ->
+  c = capture
+  -- shorthand for lexer.word
+  ident = (alpha + '_')^1 * (alpha + digit + '_')^0
+
+  -- Comments.
+  line_comment = P'//' * scan_until eol
+  block_comment = span '/*', '*/'
+  comment = c 'comment', any {line_comment, block_comment}
+
+  -- Strings.
+  dq_str = span '"', '"', '\\'
+  raw_str = span '#"', '#'
+  string  = c 'string', any {dq_str, raw_str}
+
+
+  -- Numbers.
+  number = c 'number', any {
+    float,
+    hexadecimal,
+    octal,
+    '0b' * (R'01' + '_')^1,
+    (digit + '_')^1 -- decimal integer
+  }
+
+  -- Keywords.
+  keyword = c 'keyword', word {
+    'abstract',   'alignof',    'as',       'become',   'box',
+    'break',      'const',      'continue', 'crate',    'do',
+    'else',       'enum',       'extern',   'false',    'final',
+    'fn',         'for',        'if',       'impl',     'in',
+    'let',        'loop',       'macro',    'match',    'mod',
+    'move',       'mut',        "offsetof", 'override', 'priv',
+    'proc',       'pub',        'pure',     'ref',      'return',
+    'Self',       'self',       'sizeof',   'static',   'struct',
+    'super',      'trait',      'true',     'type',     'typeof',
+    'unsafe',     'unsized',    'use',      'virtual',  'where',
+    'while',      'yield'
+  }
+
+  -- Library Types.
+  library = upper^1 * (lower + digit)^1
+
+  -- Lifetimes.
+  lifetime = "'" * ident
+
+  -- Primitive Types.
+  primitive = word {
+    'bool', 'isize', 'usize', 'char', 'str',
+    'u8', 'u16', 'u32', 'u64', 'i8', 'i16', 'i32', 'i64',
+    'f32','f64',
+  }
+
+  type = c 'type', any {library, lifetime, primitive}
+
+  -- Identifiers.
+  identifier = c 'identifier', ident
+
+  -- Operators.
+  operator = c 'operator', S'+-/*%<>!=`^~@&|?#~:;,.()[]{}'
+
+  -- Attributes.
+  attribute = c 'attribute', (P'#![' + P'#[') * scan_until(eol + P']')
```

Comment:
This confuses me a little. For starters, it won't highlight the closing brace (I think using `scan_to` instead of `scan_until` will fix that). But...why is the `eol +` there? The `eol` will always match, so lpeg will never even bother to try the `P']'`. My guess is that you were trying to highlight it correctly, even if the attribute was still being typed; in that case, I think `P']' + eol` would work better.